### DISCUSSION TITLE
What's the recommended way to filter by a null relation?

### DISCUSSION THREAD

**Discussion by jwld** [prisma/prisma#3142] [Q&A] [answered]
Can't find anything in the docs on this.

Say I have bookings and workers - each booking can be assigned to a worker.  The schema looks like this:
```prisma
model Booking {
  worker         Worker?        @relation(fields: [workerId], references: [id])
  workerId       String?
}
```

If I want to get all bookings that are not assigned to a worker, which of the following is the recommended practice?

A) Use the relation
```ts
const unassignedBookings = await prisma.booking({
  where: { worker: undefined }
})
```

B) Use the relation field
```ts
const unassignedBookings = await prisma.booking({
  where: { workerId: undefined }
})
```

**Comment by ryands17** [comment, marked as answer]
Hey @jwld 👋 

I think this would be a better approach:

```ts
const res = await prisma.booking.findMany({
    where: {
      workerId: null,
    },
  })
```

As the relation field will actually be `null` if a booking isn't assigned to a worker.

**Reply by jwld** [reply]
Oops yes I meant to put null for the second one. Ok so should we always assume it's better to filter off the relation field rather than the relation itself them?

**Reply by ryands17** [reply]
I don't think it would matter, as both would return the same dataset in this case. I think it would be according to preference as the query generated is the same :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
